Docs Homepage

mwvd opened the pull request:
[Closes #31]

## Docs homepage updates

This PR adds a refreshed homepage to the documentation site

### Changes

- Raw HTML added to render in `core/index.rst`
	- Inlined CSS
	- Homepage specific style override
		- Hides the `sphinx` generated page header for `core/index.rst`
		- Override style added to `shared/_static/home.css`
- Images added to `core/_static/assets/home`

### Notes

Reusable sections in raw generated HTML here could possibly also be automated with Javascript:

> [!NOTE]
> In this example I moved image assets from `core/_static/assets/home` into `core/_static/home/assets`
> (see `core/_static/home/data.json` below)

<details>
<summary><code>core/index.rst</code>:</summary>
<br/>

```html
<!-- ... -->
<body>
  <div class="container" id="container">
    <h1 class="hx">Tenstorrent<br/>Documentation</h1>
    <!-- sections appended here -->
  </div>
  <script async>
    const main = async () => {
      const container = document.getElementById("container");

      const response = await fetch("_static/data.json");
      const sections = await response.json();

      sections.forEach((section) => {
        const sectionElement = document.createElement("div");
        sectionElement.classList.add("section");

        const sectionTitle = document.createElement("h2");
        sectionTitle.textContent = section.title;
        sectionElement.appendChild(sectionTitle);

        const grid = document.createElement("div");
        grid.classList.add("grid");
        sectionElement.appendChild(grid);

        section.cards.forEach((card) => {
          const cardElement = document.createElement("a");
          cardElement.href = card.url;
          if (card.url.startsWith("http")) {
            cardElement.target = "_blank";
          }
          cardElement.classList.add("card");
          if (card.small) {
            cardElement.classList.add("small-card");
          }

          const image = document.createElement("img");
          image.src = card.image;
          cardElement.appendChild(image);

          const title = document.createElement(card.small ? "h4" : "h3");
          title.textContent = card.title;
          cardElement.appendChild(title);

          grid.appendChild(cardElement);
        });

        container.appendChild(sectionElement);
      });
    }
    main();
  </script>
</body>
<!-- ... -->
```

</details>

<details>
<summary><code>core/_static/home/data.json</code>:</summary>
<br/>

```json
[
  {
    "title": "Add-in Boards",
    "cards": [
      {
        "title": "Grayskull™",
        "image": "_static/home/grayskull.png",
        "url": "aibs/grayskull.html"
      },
      {
        "title": "Wormhole™",
        "image": "_static/home/wormhole.png",
        "url": "aibs/wormhole.html"
      },
      {
        "title": "Warp 100 Bridge",
        "image": "_static/home/warp-100-bridge.png",
        "url": "aibs/ack.html",
        "small": true
      },
      {
        "title": "Active Cooling Kit",
        "image": "_static/home/active-cooling-kit.png",
        "url": "aibs/warp100.html",
        "small": true
      }
    ]
  },
  {
    "title": "Systems",
    "cards": [
      {
        "title": "TT-QuietBox",
        "image": "_static/home/tt-quietbox.png",
        "url": "systems/quietbox.html"
      },
      {
        "title": "TT-LoudBox",
        "image": "_static/home/tt-loudbox.png",
        "url": "systems/t3000.html"
      },
      {
        "title": "T7000",
        "image": "_static/home/t7000.png",
        "url": "systems/t7000.html",
        "small": true
      },
      {
        "title": "T1000",
        "image": "_static/home/t1000.png",
        "url": "systems/t1000.html",
        "small": true
      }
    ]
  },
  {
    "title": "Software",
    "cards": [
      {
        "title": "TT-NN",
        "image": "_static/home/tt-nn.png",
        "url": "https://tenstorrent.github.io/ttnn/latest/index.html",
        "small": true
      },
      {
        "title": "TT-Buda™",
        "image": "_static/home/tt-buda.png",
        "url": "https://tenstorrent.github.io/pybuda/latest/index.html",
        "small": true
      },
      {
        "title": "TT-Metalium™",
        "image": "_static/home/tt-metalium.png",
        "url": "https://tenstorrent.github.io/tt-metalium/latest/index.html",
        "small": true
      },
      {
        "title": "Software/Utilities",
        "image": "_static/home/software-utilities.png",
        "url": "https://tenstorrent.github.io/syseng/latest/index.html",
        "small": true
      },
      {
        "title": "TT-MLIR",
        "image": "_static/home/tt-mlir.png",
        "url": "https://docs.tenstorrent.com/tt-mlir/",
        "small": true
      }
    ]
  },
  {
    "title": "Support",
    "cards": [
      {
        "title": "FAQ",
        "image": "_static/home/faq.png",
        "url": "https://tenstorrent.com/faq",
        "small": true
      },
      {
        "title": "Compliance",
        "image": "_static/home/compliance.png",
        "url": "support/README.html",
        "small": true
      },
      {
        "title": "Contact Us",
        "image": "_static/home/contact-us.png",
        "url": "https://tenstorrent.com/contact",
        "small": true
      }
    ]
  },
  {
    "title": "All Documentation",
    "cards": []
  }
]
```

</details>

This approach is probably easier to maintain long-term, but I don't really like the idea of introducing client-side rendering to a docs site, especially when it's async (`sections` could be inline JS and we could drop the `async`, but then it's not quickly openable from a code editor). I'm open to other thoughts here.

### Screenshots

![Screenshot 2024-11-05 at 11 49 31 AM](https://github.com/user-attachments/assets/7e59c9b9-adec-4a7c-abda-0da989da22ee)
![Screenshot 2024-11-05 at 11 49 43 AM](https://github.com/user-attachments/assets/d66a7d30-f9df-4b69-8ee9-10c17fd7d83a)
![Screenshot 2024-11-05 at 11 50 04 AM](https://github.com/user-attachments/assets/9358c826-a2ff-4d1d-9b63-6db870496cf9)
![Screenshot 2024-11-05 at 11 50 13 AM](https://github.com/user-attachments/assets/78dee8fc-9ff5-4d1b-9a4e-a4180cabf030)
![Screenshot 2024-11-05 at 11 50 30 AM](https://github.com/user-attachments/assets/cbe39c8c-24bf-4efc-a426-9ec6511a4b7d)

dimitri-tenstorrent commented:
@mwvd I have deployed this version onto staging here  https://tenstorrent-docs-test.github.io/

One of the new changes the team requested is a prominent Getting Started section on the new homepage. The new getting started section is here - https://tenstorrent-docs-test.github.io/quickstart.html

Also, we decided to remove `software/utilities box` as it was conflicting with the instructions from Getting Started.

dimitri-tenstorrent commented on core/index.rst:
```diff
@@ -1,6 +1,256 @@
 Tenstorrent
 ==============================
 
+.. raw:: html
```
Could we use `:include:` tag here and have the HTML inside of `home.html`?

mwvd replied:
Yes- much better idea- I'll do this

mwvd replied:
Couldn't figure this out with `:include:`, but was able to move the HTML into `/core/_templates/home.html` and then include it here with:

```rst
.. raw:: html
   :file: _templates/home.html
```

dimitri-tenstorrent commented on shared/_static/home.css:
```diff
@@ -0,0 +1,8 @@
+  Hides sphinx generated homepage header (https://docs.tenstorrent.com/)
+*/
+
+#tenstorrent > h1 {
```
Can I not remove it via deleting the first two lines of `index.rst` 
```
Tenstorrent
==============================
```

mwvd replied:
> ```
> Tenstorrent
> ==============================
> ```

Removing this seems to make the page title default to the closest header:

![Screenshot 2024-11-25 at 10 56 50 AM](https://github.com/user-attachments/assets/cf0c92ef-9d7b-4e5d-adb7-288c22b1a816)

dimitri-tenstorrent reviewed (commented):
Reviewed 👍

mwvd commented:
@dimitri-tenstorrent seems like the staging build is missing everything below the "All Documentation" header:

![image](https://github.com/user-attachments/assets/90f12a7b-f0fd-4d94-9e9c-f5ba5f3dfc56)

Is this just because of config handling orchestration for staging? Or would this be the same in the production environment also

mwvd commented:
> One of the new changes the team requested is a prominent Getting Started section on the new homepage. The new getting started section is here - https://tenstorrent-docs-test.github.io/quickstart.html

I'll talk to Keith about this tomorrow and aim for EOW in another PR

> Also, we decided to remove software/utilities box as it was conflicting with the instructions from Getting Started.
✅

dimitri-tenstorrent commented:
> @dimitri-tenstorrent seems like the staging build is missing everything below the "All Documentation" header:
> 
> ![image](https://private-user-images.githubusercontent.com/4127151/387410688-90f12a7b-f0fd-4d94-9e9c-f5ba5f3dfc56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5Njg5MjEsIm5iZiI6MTczMTk2ODYyMSwicGF0aCI6Ii80MTI3MTUxLzM4NzQxMDY4OC05MGYxMmE3Yi1mMGZkLTRkOTQtOWU5Yy1mNWJhNWYzZGZjNTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMjIyMzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTk2NDJmMDEwZThmOGVjMTJiZWNhYmI4YmI3NTI2ZDc5YjkwOWMyYjVlZDNjMTVhY2U2OTRmZjZiODhjMDM1YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FYWPlSwDXXI2dHCGMOI6d2lF3_9d41TjEjBZaT3kwdI)
> 
> Is this just because of config handling orchestration for staging? Or would this be the same in the production environment also

Ah, the table of contents was hidden in the last version, but I can re-add it to this one.

dimitri-tenstorrent commented:
@mwvd I am picking this up again with the aim to deploy the homepage updates you have made.
Remind me if we already discussed these scrollbars that appear on the page - see screenshot
![Screenshot 2025-01-13 at 4 24 41 PM](https://github.com/user-attachments/assets/9b8dcb67-f82b-4ab5-92bb-702a96c3cda1)

mwvd commented:
> @mwvd I am picking this up again with the aim to deploy the homepage updates you have made. Remind me if we already discussed these scrollbars that appear on the page - see screenshot ![Screenshot 2025-01-13 at 4 24 41 PM](https://private-user-images.githubusercontent.com/119051828/402703332-9b8dcb67-f82b-4ab5-92bb-702a96c3cda1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4ODA5NjQsIm5iZiI6MTczNjg4MDY2NCwicGF0aCI6Ii8xMTkwNTE4MjgvNDAyNzAzMzMyLTliOGRjYjY3LWY4MmItNGFiNS05MmJiLTcwMmE5NmMzY2RhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNFQxODUxMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzQ0YTdhODRkYWNjMWYwNmUzOWUzOWM4ZDU2ZjNmODdhNmI0MzJlZjJmMmY1NWNmOGQ3MTg2YmQxNDZjZTA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EDTDSiuF6hZNx5KNNzsP45z1V4UYhyNR4XsX5YLfsuM)

@dimitri-tenstorrent not seeing these on my end, but I think I know what might be causing- I'll investigate and fix in the next few days.

Thanks!

mwvd commented on core/_templates/home.html:
```diff
@@ -142,6 +144,15 @@
         font-size: 144px;
       }
     }
+
```
@dimitri-Tenstorrent

This change should fix the scroll bars appearing. These scrollbars show when users have "Show scroll bars" set to "Always" (on macOS - imagine Windows is similar but unable to test).

In [`shared/_static/tt-theme.css`](https://github.com/tenstorrent/tenstorrent.github.io/blob/c07709b98d98096b9af8c694057e252209674c45/shared/_static/tt_theme.css#L360-L370), `overflow-x: scroll` is set for header elements:

```css
/* Document styling */

.document * h1,
.document * h2,
.document * h3,
.document * h4 {
  /* vertically center document header links */
  display: flex;
  align-items: center;
  gap: 0.5rem;
  max-width: 100%;
  overflow-x: scroll; /* here */
}
```

This override prevents the above from applying scrollbars to header elements. When you have a chance, could you pull and take a look at this, please?

Let me know if you're still seeing any issues on your end.

dimitri-tenstorrent replied:
Deployed to staging :) https://tenstorrent-docs-test.github.io/

dimitri-tenstorrent replied:
@mwvd Not seeing the issue anymore.

mwvd commented:
@dimitri-Tenstorrent made the following changes:

- added a section at the top of the home page that links to a new `/getting-started` page
- moved `quickstart.md` into `/getting-started` and renamed to `README.md`
- updated TOC tree and links where applicable

dimitri-tenstorrent commented:
Deployed to staging - https://tenstorrent-docs-test.github.io/index.html

mwvd commented:
> Deployed to staging - [tenstorrent-docs-test.github.io/index.html](https://tenstorrent-docs-test.github.io/index.html)

@dimitri-Tenstorrent @dsklavos - looks good to me - what do you two think?

Are we okay with the routes / URLs here?
